connectors/sources/google_drive.py

```diff
@@ -157,12 +171,15 @@ async def list_files(self, fetch_permissions=False):
             if fetch_permissions
             else DRIVE_ITEMS_FIELDS
         )
-
+        if last_sync_time is None:
+            list_query = "trashed=false"
+        else:
+            list_query = f"trashed=true or modifiedTime > '{last_sync_time}'"
```

Comment:
Why do we specify explicitly `trashed=true or ...` 

This would fetch all trashed files in each incremental sync (potentially large number of trashed files). 

Could we instead omit `trashed` from query `q` and just use  `modifiedTime > '{last_sync_time}'` in query. 

And then check if a file was recently trashed (and if so, mark it as deleted) while iterating over response from that query?